dev/PipsPager/PipsPager.cpp

```diff
@@ -460,9 +473,49 @@ void PipsPager::OnOrientationChanged()
     {
         winrt::VisualStateManager::GoToState(*this, c_pipsPagerVerticalOrientationVisualState, false);
     }
+    if (const auto repeater = m_pipsPagerRepeater.get())
+    {
+        if (const auto itemsSourceView = repeater.ItemsSourceView())
+        {
+            const auto itemCount = itemsSourceView.Count();
+            for (int i = 0; i < itemCount; i++)
+            {
+                if (const auto pip = repeater.TryGetElement(i).try_as<winrt::Control>())
+                {
+                    UpdatePipOrientation(pip);
+                }
+            }
+        }
+    }
+    m_defaultPipSize = GetDesiredPipSize(NormalPipStyle());
+    m_selectedPipSize = GetDesiredPipSize(SelectedPipStyle());
     SetScrollViewerMaxSize();
-    UpdateSelectedPip(SelectedPageIndex());
+    if (const auto selectedPip = GetSelectedItem())
+    {
+        ScrollToCenterOfViewport(selectedPip, SelectedPageIndex());
+    }
+}
 
+void PipsPager::ApplyStyleToPipAndUpdateOrientation(const winrt::FrameworkElement& pip, const winrt::Style& style)
+{
+    pip.Style(style);
+    if (const auto control = pip.try_as<winrt::Control>())
+    {
+        control.ApplyTemplate();
```

Comment:
This seems expensive... This function gets called every time the selection changes and will reapply the template.. Why is that needed?

Reply:
Because every time selection changes - we need to update the style of new Selected pip and old Selected pip. When we do that, their previous orientation visual states are not active anymore because we switched the template, so we have to do it again. Would you suggest a better way to handle it?